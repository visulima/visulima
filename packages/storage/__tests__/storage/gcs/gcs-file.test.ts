import { describe, expect, it } from "vitest";

import GCSFile from "../../../src/storage/gcs/gcs-file";

describe(GCSFile, () => {
    it("should include GCS-specific properties and match snapshot", () => {
        expect.assertions(1);

        const file = new GCSFile({ metadata: { foo: "bar" } });

        file.GCSUploadURI = "https://storage.googleapis.com/upload/example";
        file.uri = "gs://bucket-name/file-name";

        file.id = "GCSFileTestId123456789";

        file.originalName = "GCSFileTestId123456789";

        expect(file).toMatchSnapshot();
    });
});
