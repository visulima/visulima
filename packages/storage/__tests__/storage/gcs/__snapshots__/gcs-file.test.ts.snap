// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GCSFile > should include GCS-specific properties and match snapshot 1`] = `
GCSFile {
  "ETag": undefined,
  "GCSUploadURI": "https://storage.googleapis.com/upload/example",
  "bytesWritten": NaN,
  "content": undefined,
  "contentType": "application/octet-stream",
  "createdAt": undefined,
  "expiredAt": undefined,
  "hash": undefined,
  "id": "GCSFileTestId123456789",
  "metadata": {
    "foo": "bar",
  },
  "modifiedAt": undefined,
  "name": "",
  "originalName": "GCSFileTestId123456789",
  "size": undefined,
  "status": undefined,
  "uri": "gs://bucket-name/file-name",
}
`;
